tests/Common/TriggersTest.php

```diff
@@ -0,0 +1,171 @@
+            'tableIds' => ['nothing-is-here'],
+        ];
+        unset($data[$keyToDelete]);
+        $this->expectExceptionMessage(sprintf('Missing required query parametr(s) [%s]', $keyToDelete));
+        $this->expectException(ClientException::class);
+        $this->_client->createTrigger($data);
+    }
+
+    public function deleteKeyProvider(): array
+    {
+        return [
+            ['component'],
+            ['configurationId'],
+            ['coolDownPeriod'],
+            ['runWithTokenId'],
+            ['tableIds'],
+        ];
+    }
+
+    public function testDeleteTrigger(): void
+    {
+        $table = $this->createTableWithRandomData("watched-2");
+        $newTokenId = $this->_client->createToken([$this->getTestBucketId() => 'read']);
+
+        $trigger = $this->_client->createTrigger([
+            'component' => 'orchestrator',
+            'configurationId' => 123,
+            'coolDownPeriod' => 10,
+            'runWithTokenId' => $newTokenId,
+            'tableIds' => [
+                $table,
+            ],
+        ]);
+
+        $loadedTrigger = $this->_client->getTrigger($trigger['id']);
+        $this->assertEquals($trigger['id'], $loadedTrigger['id']);
+
+        $this->_client->deleteTrigger($loadedTrigger['id']);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('Trigger with id [%d] was not found', $loadedTrigger['id']));
+        $this->_client->getTrigger($trigger['id']);
+    }
+
+    public function testListAction(): void
+    {
+        $table = $this->createTableWithRandomData("watched-2");
+        $newTokenId = $this->_client->createToken([$this->getTestBucketId() => 'read']);
+
+        $trigger1ConfigurationId = time();
+        $componentName = uniqid('test');
+        $trigger1 = $this->_client->createTrigger([
+            'component' => $componentName,
+            'configurationId' => $trigger1ConfigurationId,
+            'coolDownPeriod' => 10,
+            'runWithTokenId' => $newTokenId,
+            'tableIds' => [
+                $table,
+            ],
+        ]);
+        $trigger2 = $this->_client->createTrigger([
+            'component' => 'keboola.ex-manzelka',
+            'configurationId' => 123,
+            'coolDownPeriod' => 10,
+            'runWithTokenId' => $newTokenId,
+            'tableIds' => [
+                $table,
+            ],
+        ]);
+
+        $triggers = $this->_client->listTriggers();
+        $trigger1Found = $trigger2Found = false;
+        foreach ($triggers as $trigger) {
+            if ($trigger1['id'] == $trigger['id']) {
+                $trigger1Found = true;
+            }
+            if ($trigger2['id'] == $trigger['id']) {
+                $trigger2Found = true;
+            }
+        }
+
+        $this->assertTrue($trigger1Found);
+        $this->assertTrue($trigger2Found);
+
+        $triggers = $this->_client->listTriggers(
```

Comment:
Splácnul jsem tady víc věcí dohromady, protože to vytváření tabulek trvá docela dlouho a tak aby se ušetřil nějaký čas. Když tak to rozdělím na více testů.